client/rest/src/main/java/org/elasticsearch/client/RestClient.java

```diff
@@ -560,75 +606,129 @@ private void setHeaders(HttpRequest httpRequest, Collection<Header> requestHeade
     }
 
     /**
-     * Returns an {@link Iterable} of hosts to be used for a request call.
-     * Ideally, the first host is retrieved from the iterable and used successfully for the request.
-     * Otherwise, after each failure the next host has to be retrieved from the iterator so that the request can be retried until
-     * there are no more hosts available to retry against. The maximum total of attempts is equal to the number of hosts in the iterable.
-     * The iterator returned will never be empty. In case there are no healthy hosts available, or dead ones to be be retried,
-     * one dead host gets returned so that it can be retried.
+     * Returns a non-empty {@link Iterator} of nodes to be used for a request
+     * that match the {@link NodeSelector}.
+     * <p>
+     * If there are no living nodes that match the {@link NodeSelector}
+     * this will return the dead node that matches the {@link NodeSelector}
+     * that is closest to being revived.
+     * <p>
+     * If no living and no dead nodes match the selector we retry a few
+     * times to handle concurrent modifications of the list of dead nodes.
```

Comment:
I think that we don't retry a few times anymore as we read the blacklist only once and work with that version. can you adapt the comment?